plugins/microservices/msi_get_file_descriptor_info/libmsi_get_file_descriptor_info.cpp

```diff
@@ -0,0 +1,81 @@
+/// \file
+
+#include "irods_ms_plugin.hpp"
+#include "irods_re_structs.hpp"
+#include "msParam.h"
+#include "rodsErrorTable.h"
+#include "irods_error.hpp"
+#include "irods_log.hpp"
+#include "rs_get_file_descriptor_info.hpp"
+
+#include "json.hpp"
+
+#include <functional>
+#include <string>
+
+namespace
+{
+    using json = nlohmann::json;
+
+    int msi_impl(msParam_t* _in, msParam_t* _out, ruleExecInfo_t* _rei)
+    {
+        // check sanity
+        if (!_in || !_out) {
+            return SYS_INTERNAL_NULL_INPUT_ERR;
+        }
+
+        // we want a positive index
+        int index = 0;
+
+        if (strcmp(_in->type, INT_MS_T) || (index = parseMspForPosInt(_in)) < 0) {
+            return SYS_INVALID_INPUT_PARAM;
+        }
+
+        // get file descriptor info API
+        json input{{"fd", index}};
+        char *output = nullptr;
+        int status = 0;
+
+        if ((status = rs_get_file_descriptor_info(_rei->rsComm,
+                                                  input.dump().c_str(),
+                                                  &output)) < 0) {
+            return status;
+        }
```

Comment:
given that `status` is not used after the if-statement, this can be tighter. for example:
```c++
if (const auto ec = rs_get_file_descriptor_info(_rei->rsComm, input.dump().c_str(), &output); ec < 0) {
    return ec;
}
```
also, doing an assignment and comparison like on lines 39 through 41 can be misread very easily. i'd recommend either using the C++17 syntax or separating the function call from the error code check.

Reply:
If you think so. Fixed.